### PR TITLE
✨ Register a new user on the Register page

### LINKED ISSUE #8: Register
As a new user I want the option to create a new account.

----
- [ ] Add link to register form

### PR DESCRIPTION
**Pull request by phlgr**
Fix #8 
Similar to the `useFetch` custom hook, `useMutation` is used to send data.
It's fully typed (the TS part is over 9000, not easy to understand), but the logic/JS is worth to investigate 🕵️‍♀️.
The implementation is inspired by https://react-query.tanstack.com/reference/useMutation

You can test the register page on https://photoplay900-registerus-qfarxu.herokuapp.com/register.
The submit button is disabled if you submit the form to prevent multiple requests (throttle your connection to see it).
Error messages are displayed next to the button (email already exists, passwords didn't match).
If the registration succeeded, the user is redirected to the home page.

Adding a profile picture is not implemented yet.

### CODE REVIEW

File: src/app/pages/Register/Register.tsx [hunk 1]
```diff
@@ -1,20 +1,30 @@
...
   const [lastName, setLastName] = useState('');
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
   const [confirmPassword, setConfirmPassword] = useState('');
 
-  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
     event.preventDefault();
-    alert(`${firstName} submitted`);
+    const user = { email, firstName, lastName, password };
```
Comment (lmachens) [inline src/app/pages/Register/Register.tsx]: Good job. I would feel more safe if you add `User` type here:
https://github.com/lmachens/photoplay/blob/main/src/types.ts#L27

```suggestion
    const user: User = { email, firstName, lastName, password };
```

File: src/app/pages/Register/Register.tsx [hunk 2]
```diff
@@ -1,20 +1,30 @@
...
+    await fetch('/api/users', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(user),
+    });
```
Comment (lmachens) [inline src/app/pages/Register/Register.tsx]: What happens when the `email` already exists or a different error is thrown?
Can you catch errors?

**Review by lmachens — COMMENTED** [review]
Good job. It works well, but not in every case. 
I am fine with this solution if you add a follow-up task to add error catching, loading state and custom hooks.

I would target a solution like this:

(simplified)
```tsx
function RegisterForm(): JSX.Element {
  const { isLoading, error, mutate } = useMutation(postUser, {
    onSuccess: () => {
      history.push('/');
    }
  });
  
  function handleSubmit(event) {
    event.preventDefault();
    const user: User = { ... };
    mutate(user);
  }

  return (
    <form onSubmit={handleSubmit}>
      <button disabled={isLoading}>Submit</button>
      {error && <div>ERROR! {error}</div>}
    </form>
  );
}
```

What do you think?

**Review by Dev-dfm — APPROVED** [review]
lgtm 💯  Thanks!

**Review by daebert — APPROVED** [review]
lgtm!

**Comment by hugh-burgess** [comment]
LGTM! I was able to create a password with just spaces so that could be a bug. But otherwise very nice layout!

**Review by lajanzen — APPROVED** [review]
🚀

**Review by infoucarsinan — APPROVED** [review]

**Pull request merged**